src/utils/include/settings.h

```diff
@@ -0,0 +1,68 @@
+#ifndef OPENDIGITIZER_SETTINGS_H
+#define OPENDIGITIZER_SETTINGS_H
+
+#include "RestClient.hpp"
+#include <algorithm>
+#include <ranges>
+#include <string>
+#include <vector>
+
+namespace Digitizer {
+
+namespace detail {
+
+template<typename T>
+inline auto toType(std::string string) {
+    return T{string};
+}
+template<typename T>
+requires std::is_arithmetic_v<T>
+inline T toType(std::string string) {
+    return static_cast<T>(std::stod(string.c_str()));
+}
+}
+
+template<typename T>
+inline T getValueFromEnv(std::string variableName, T defaultValue) {
+    if (auto valueString = std::getenv(variableName.c_str()); valueString) {
+        return detail::toType<T>(valueString);
+    } else {
+        return defaultValue;
+    }
+}
+
+template<>
+inline bool getValueFromEnv(std::string variableName, bool defaultValue) {
+    if (auto value = std::getenv(variableName.c_str()); value) {
+        if (std::ranges::any_of(std::vector<std::string>{ "0", "false", "False", "none" }, [value](auto a) { return a == std::string{ value }; })) {
+            return false;
+        } else if (std::ranges::any_of(std::vector<std::string>{ "1", "true", "True", "yes" }, [value](auto a) { return a == std::string{ value }; })) {
+            return true;
+        }
+    }
+    return defaultValue;
+}
+
+struct Settings {
+    Settings() {
+        disableHttps                                    = getValueFromEnv("DISABLE_HTTPS", disableHttps);
+        checkCertificates                               = getValueFromEnv("CHECK_CERTIFICATES", checkCertificates);
+        hostname                                        = getValueFromEnv("DIGITIZER_HOSTNAME", hostname);
+        port                                            = getValueFromEnv("DIGITIZER_PORT", port);
+
+        opencmw::client::RestClient::CHECK_CERTIFICATES = false;
```

Comment:
Changing a static global of rest client inside of the Settings constructor doesn't look like the nice way, imo it should just be moved  to the application setup and explicitly set from the settings object there. It can then be later refactored to have the client access the settings object.

Reply:
it should definitely use the `checkCertificates` value instead of just setting false :man_facepalming: 
but it does make sense to me to actually set the value that we represent here, one could argue that if `Settings` were passed through the stack or implemented as a singleton, it should also get a setter for `checkCertificate` which sets `CHECK_CERTIFICATES`
I don't think we could refactor the code to have the client access this setting, because that is happening in opencmw through the static CHECK_CERTIFICATES.

But i can also just move it to the application start for now.